### PR TITLE
Fix SORT in subqueries when crossing block boundaries

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

SORT inside a subquery could sometimes swallow part of its input when it crossed block boundaries.

- [X] :hankey: Bugfix 
- [x] :book: CHANGELOG entry made
- [X] :muscle: The behavior in this PR was *manually tested*
- [x] :computer: The behavior change can be verified via automatic tests

#### Backports:

- [X] Backport required for: *3.7*

#### Related Information

- [X] GitHub issue: https://github.com/arangodb/arangodb/issues/12693

### Testing & Verification

- [X] This change is already covered by existing tests, such as `InputRangeTest` in the gtest suite `arangodbtest`.
- [x] This PR adds tests that were used to verify all changes:
  - [x] Added **Regression Tests**

### Documentation

- [x] Added a *Changelog Entry* (referencing the corresponding public or internal issue number)

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/12080/

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/12170/

File: arangod/Aql/AqlItemMatrix.h [hunk 1]
```diff
@@ -73,7 +73,7 @@ class AqlItemMatrix {
    *
    * @return True if empty
    */
-  bool empty() const noexcept;
+  bool blocksEmpty() const noexcept;
```
Comment (mchacki) [inline arangod/Aql/AqlItemMatrix.h:76]: Name is better :+1:

File: tests/Aql/InputRangeTest.cpp [hunk 2]
```diff
@@ -136,8 +135,28 @@ TYPED_TEST_CASE_P(InputRangeTest);
 TYPED_TEST_P(InputRangeTest, test_default_initializer) {
   std::vector<ExecutorState> states{ExecutorState::DONE, ExecutorState::HASMORE};
   for (auto const& finalState : states) {
+    if (std::is_same_v<AqlItemBlockInputMatrix, TypeParam> &&
+        finalState == ExecutorState::DONE) {
+      // The AqlItemBlockInputMatrix may not be instantiated with DON
```
Comment (mchacki) [inline tests/Aql/InputRangeTest.cpp:140]:
```suggestion
      // The AqlItemBlockInputMatrix may not be instantiated with DONE
```

**Review by mchacki — APPROVED** [review]
LGTM

**Pull request merged**